### PR TITLE
Stop ruleset selector from propagating positional input

### LINKED ISSUE #3548: Clicking the (disabled) gamemode switch buttons in score select crashes the game
Using 2018.1004.1, clicking one of the gamemode switches (after completing a osu! map, and clicking osu!taiko for example), brings an error, clicking it 2 times more will actually crash the game.

### PR DESCRIPTION
**Pull request by plankp**
Fix #3548

### CODE REVIEW

**Comment by peppy** [comment]
Probably want to investigate how this regressed before blindly merging, since I don't think this crash used to exist.

**Comment by plankp** [comment]
The issue only begin to appear in 2018.1004.0 Going to try to find what actually happened.

**Comment by smoogipoo** [comment]
The reason for this is https://github.com/ppy/osu/pull/3523 . `ToolbarRulesetSelector` should do `PropagatePositionalInputSubTree => false` instead of `HandlePositionalInput => false`.

It's worth going through the PR I mentioned and verifying any similar cases have been converted over appropriately.

**Comment by plankp** [comment]
Also fixes #3562

File: osu.Game/Overlays/Toolbar/ToolbarRulesetSelector.cs [hunk 1]
```diff
@@ -105,6 +105,8 @@ protected override bool OnKeyDown(KeyDownEvent e)
         public override bool HandleNonPositionalInput => !ruleset.Disabled && base.HandleNonPositionalInput;
         public override bool HandlePositionalInput => !ruleset.Disabled && base.HandlePositionalInput;
 
+        public override bool PropagatePositionalInputSubTree => !ruleset.Disabled;
```
Comment (peppy) [inline osu.Game/Overlays/Toolbar/ToolbarRulesetSelector.cs]: `&& base.` for safety (see two lines above)?

**Review by peppy — CHANGES_REQUESTED** [review]

**Comment by plankp** [comment]
I think somewhere between github and appveyor broke...

**Review by peppy — APPROVED** [review]

**Pull request merged**